src/form.tsx

```diff
@@ -134,6 +139,29 @@ export function Form<FV extends FormValues>({
         [onSubmit, extCtx]
     )
 
+    // Ripped from here: https://www.wpeform.io/blog/exit-prompt-on-window-close-react-app/
+    useEffect(() => {
+        if (isSSR || !confirmOnNavigate) return
+
+        // the handler for actually showing the prompt
+        // https://developer.mozilla.org/en-US/docs/Web/API/WindowEventHandlers/onbeforeunload
+        const handler = (event: BeforeUnloadEvent) => {
+            event.preventDefault()
+            event.returnValue = ''
+        }
+
+        if (fc.formState.isDirty) {
+            window.addEventListener('beforeunload', handler)
+            // clean it up, if the dirty state changes
+            return () => {
+                window.removeEventListener('beforeunload', handler)
+            }
+        }
+        // since this is not dirty, don't do anything
+        // eslint-disable-next-line @typescript-eslint/no-empty-function
+        return () => {}
+    }, [fc.formState.isDirty, confirmOnNavigate])
+
```

Comment:
I don't think this will do what you expect.  `beforeunload` is only triggered on a full page navigation, not on a history.push SPA navigation, which is what pretty much all our apps use.   https://stackoverflow.com/questions/47583856/window-location-href-vs-history-pushstate-which-to-use

I'll make a test for you to look at shortly.